elyra/pipeline/component_registry.py

```diff
@@ -41,21 +41,31 @@ class ComponentRegistry(LoggingConfigurable):
                               op="execute-notebook-node",
                               source_type="elyra",
                               source="elyra",
-                              extensions=[".ipynb"]),
+                              extensions=[".ipynb"],
+                              category="generic"),
         "python-script": Component(id="python-script",
                                    name="Python Script",
                                    description="Run Python script",
                                    op="execute-python-node",
                                    source_type="elyra",
                                    source="elyra",
-                                   extensions=[".py"]),
+                                   extensions=[".py"],
+                                   category="generic"),
         "r-script": Component(id="r-script",
                               name="R Script",
                               description="Run R script",
                               op="execute-r-node",
                               source_type="elyra",
                               source="elyra",
-                              extensions=[".r"])}
+                              extensions=[".r"],
+                              category="generic")}
+
+    _generic_category: dict = SimpleNamespace(**({
+        "id": "generic",
+        "label": "Generic Components",
+        "image": "",
+        "description": "<GENERIC_CATEGORY_DESCRIPTION>"
```

Comment:
Is this just a placeholder or tied to some magic?

Reply:
Per SCRUM, we'll update the descriptions in the catalogs to "Components that are supported by [runtime platform]"